src/jobs/BenchmarkJob.jl

```diff
@@ -261,17 +261,34 @@ function execute_benchmarks!(job::BenchmarkJob, whichbuild::Symbol)
     cd(builddir)
 
     # update local Julia packages for the relevant Julia version
-    run(`$juliapath -e 'Pkg.update()'`)
+    run(`$juliapath -e 'VERSION >= v"0.7.0-DEV.3656" && using Pkg; Pkg.update()'`)
 
     # add/update BaseBenchmarks for the relevant Julia version + use branch specified by cfg
     nodelog(cfg, node, "updating local BaseBenchmarks repo")
     branchname = cfg.testmode ? "test" : "nanosoldier"
-    oldpwd = pwd()
-    try run(`$juliapath -e 'Pkg.clone("https://github.com/JuliaCI/BaseBenchmarks.jl")'`) end
-    cd(read(`$juliapath -e 'print(Pkg.dir("BaseBenchmarks"))'`, String))
-    run(`git fetch --all --quiet`)
-    run(`git reset --hard --quiet origin/$(branchname)`)
-    cd(oldpwd)
+    try
+        run(```
+            $juliapath -e '
+                VERSION >= v"0.7.0-DEV.3656" && using Pkg
+                url = "https://github.com/JuliaCI/BaseBenchmarks.jl"
+                VERSION >= v"0.7.0-DEV.5183" ? Pkg.add(url) : Pkg.clone(url)
+            '
+            ```)
+    catch
```

Comment:
Is it really ok to ignore errors here?

Reply:
Yes this one is okay, since it will error if the package has already been added, which is fine.